jerry-core/lit/lit-internal-magic-strings.inc.h

```diff
@@ -0,0 +1,74 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LIT_INTERNAL_MAGIC_STRINGS_H
+#define LIT_INTERNAL_MAGIC_STRINGS_H
+
+#ifndef CONFIG_DISABLE_ES2015_PROMISE_BUILTIN
+/**
+ * [[Promise]] of promise reject or resolve functions
+ */
+LIT_INTERNAL_MAGIC_STRING_DEF (LIT_INTERNAL_MAGIC_STRING_PROMISE = LIT_NON_INTERNAL_MAGIC_STRING__COUNT)
+/**
+ * [[AlreadyResolved]] of promise reject or resolve functions
+ */
+LIT_INTERNAL_MAGIC_STRING_DEF (LIT_INTERNAL_MAGIC_STRING_ALREADY_RESOLVED)
+/**
+ * The resolve funtion of the promise object
+ */
+LIT_INTERNAL_MAGIC_STRING_DEF (LIT_INTERNAL_MAGIC_STRING_RESOLVE_FUNCTION)
+/**
+ * The reject function of the promise object
+ */
+LIT_INTERNAL_MAGIC_STRING_DEF (LIT_INTERNAL_MAGIC_STRING_REJECT_FUNCTION)
+/**
+ * [[Promise]] property
+ */
+LIT_INTERNAL_MAGIC_STRING_DEF (LIT_INTERNAL_MAGIC_STRING_PROMISE_PROPERTY_PROMISE)
+/**
+ * [[Resolve]] property
+ */
+LIT_INTERNAL_MAGIC_STRING_DEF (LIT_INTERNAL_MAGIC_STRING_PROMISE_PROPERTY_RESOLVE)
+/**
+ * [[Reject]] property
+ */
+LIT_INTERNAL_MAGIC_STRING_DEF (LIT_INTERNAL_MAGIC_STRING_PROMISE_PROPERTY_REJECT)
+/**
+ * [[Capability]] property
+ */
+LIT_INTERNAL_MAGIC_STRING_DEF (LIT_INTERNAL_MAGIC_STRING_PROMISE_PROPERTY_CAPABILITY)
+/**
+ * [[Handler]] property
+ */
+LIT_INTERNAL_MAGIC_STRING_DEF (LIT_INTERNAL_MAGIC_STRING_PROMISE_PROPERTY_HANDLER)
+/**
+ * [[AlreadyCalled]] property
+ */
+LIT_INTERNAL_MAGIC_STRING_DEF (LIT_INTERNAL_MAGIC_STRING_PROMISE_PROPERTY_ALREADY_CALLED)
+/**
+ * [[Index]] property
+ */
+LIT_INTERNAL_MAGIC_STRING_DEF (LIT_INTERNAL_MAGIC_STRING_PROMISE_PROPERTY_INDEX)
+/**
+ * [[Values]] property
+ */
+LIT_INTERNAL_MAGIC_STRING_DEF (LIT_INTERNAL_MAGIC_STRING_PROMISE_PROPERTY_VALUE)
+/**
+ * [[RemainingElement]] property
+ */
+LIT_INTERNAL_MAGIC_STRING_DEF (LIT_INTERNAL_MAGIC_STRING_PROMISE_PROPERTY_REMAINING_ELEMENT)
+#endif /* !CONFIG_DISABLE_ES2015_PROMISE_BUILTIN */
+
+#endif /* !LIT_INTERNAL_MAGIC_STRINGS_H */
```

Comment:
Just like byte codes, I would not put these behind ifdefs, because this would affect static snapshots.

Btw increase snapshot version number.